### PR TITLE
Make most ios_workflow host tool calls async

### PR DESCRIPTION
**Pull request by cbracken**
Does not yet migrate hasIdeviceId since that results in a cascade of
breaking interface changes that's significant enough for a separate
patch.

### CODE REVIEW

**Comment by cbracken** [comment]
As per @tvolkert's suggestion on #10254

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart [hunk 1]
```diff
@@ -32,65 +33,64 @@ class IOSWorkflow extends DoctorValidator implements Workflow {
...
-    if (result.isNotEmpty && !exitsHappy(<String>['idevicename']))
+    final ProcessResult result = (await runAsync(<String>['idevice_id', '-l'])).processResult;
+    if (result.exitCode == 0 && result.stdout.isNotEmpty && !await exitsHappyAsync(<String>['idevicename']))
       return false;
 
     return true;
   }
 
-  bool get hasIDeviceInstaller => exitsHappy(<String>['ideviceinstaller', '-h']);
+  Future<bool> get hasIDeviceInstaller => exitsHappyAsync(<String>['ideviceinstaller', '-h']);
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: If you only have async versions, I wouldn't put Async in the name.

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart [hunk 2]
```diff
@@ -32,65 +33,64 @@ class IOSWorkflow extends DoctorValidator implements Workflow {
...
 
-  bool get hasIosDeploy => exitsHappy(<String>['ios-deploy', '--version']);
+  Future<bool> get hasIosDeploy => exitsHappyAsync(<String>['ios-deploy', '--version']);
 
   String get iosDeployMinimumVersion => '1.9.0';
 
-  String get iosDeployVersionText => runSync(<String>['ios-deploy', '--version']).replaceAll('\n', '');
+  Future<String> get iosDeployVersionText async =>
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: nit: no line-break after =>

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart [hunk 3]
```diff
@@ -32,65 +33,64 @@ class IOSWorkflow extends DoctorValidator implements Workflow {
...
     }
   }
 
   /// Whether CocoaPods ran 'pod setup' once where the costly pods' specs are cloned.
-  bool get isCocoaPodsInitialized {
-    return fs.isDirectorySync(
-      fs.path.join(homeDirPath, '.cocoapods', 'repos', 'master')
-    );
-  }
+  Future<bool> get isCocoaPodsInitialized =>
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: ditto

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart (hunk 2 above)
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: Done.

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart (hunk 3 above)
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: Done.

**Comment by Hixie** [comment]
LGTM

**Review by tvolkert — APPROVED** [review]
LGTM!

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart (hunk 1 above)
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: The non-async version is still used elsewhere. @tvolkert is the plan to kill the sync version?

Comment (tvolkert) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: Yep

**Comment by tvolkert** [comment]
This gets us closer on #7102 - thanks!

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart (hunk 1 above)
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: Answer is yes. 60 calls plus the definition remain.

**Comment by Hixie** [comment]
When checking in when the tree is red, please make sure to include a message explaining why you feel it is justified to do so (e.g. if the failure is flaky or if the checkin is intended to fix the redness).

**Pull request merged**